### PR TITLE
fix(repository): ignore unchanged visibility error

### LINKED ISSUE #1364: [BUG] Updating repository visibility broken
**Terraform Version**
`v1.1.9` (though this has nothing to do with the terraform version)

**Provider Version**
`v5.7.0` (though I believe it is also broken in latest)

**Affected Resource(s)**
`github_repository`

**Terraform Configuration Files**
```terraform
resource "github_repository" "internal-repo-test" {
    name       = "internal-repo-test"
    visibility = "internal"
}
```

**Output**
from `terraform apply`

```terraform
Terraform will perform the following actions:

  # github_repository.internal-repo-test will be created
  + resource "github_repository" "internal-repo-test" {
      + allow_auto_merge            = false
      + allow_merge_commit          = true
      + allow_rebase_merge          = true
      + allow_squash_merge          = true
      + archived                    = false
      + default_branch              = (known after apply)
      + delete_branch_on_merge      = false
      + etag                        = (known after apply)
      + full_name                   = (known after apply)
      + git_clone_url               = (known after apply)
      + html_url                    = (known after apply)
      + http_clone_url              = (known after apply)
      + id                          = (known after apply)
      + merge_commit_message        = "PR_TITLE"
      + merge_commit_title          = "MERGE_MESSAGE"
      + name                        = "internal-repo-test"
      + node_id                     = (known after apply)
      + private                     = (known after apply)
      + repo_id                     = (known after apply)
      + squash_merge_commit_message = "COMMIT_MESSAGES"
      + squash_merge_commit_title   = "COMMIT_OR_PR_TITLE"
      + ssh_clone_url               = (known after apply)
      + svn_url                     = (known after apply)
      + visibility                  = "internal"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

github_repository.internal-repo-test: Creating...
╷
│ Error: PATCH https://api.github.com/repos/myorg/internal-repo-test: 422 Visibility can't be changed by this user. and Visibility is already internal. []
│ 
│   with github_repository.internal-repo-test,
│   on main.tf line 17, in resource "github_repository" "internal-repo-test":
│   17: resource "github_repository" "internal-repo-test" {
│ 
╵
Releasing state lock. This may take a few moments...
```

**Expected Behaviour**
Repository creation should have completed successfully.

**Actual Behaviour**
Repository creation failed. The following steps occurred:
- The code [initially creates the repository with the correct visibility](https://github.com/integrations/terraform-provider-github/blob/e827c6f14ae0ca63327bd729dab3a517b3ecc2d5/github/resource_github_repository.go#L399)
- after creating the repository, the code [updates the repository](https://github.com/integrations/terraform-provider-github/blob/e827c6f14ae0ca63327bd729dab3a517b3ecc2d5/github/resource_github_repository.go#L480)
- the update [sets the visibility to nil](https://github.com/integrations/terraform-provider-github/blob/e827c6f14ae0ca63327bd729dab3a517b3ecc2d5/github/resource_github_repository.go#L597) so that it [always updates](https://github.com/integrations/terraform-provider-github/blob/e827c6f14ae0ca63327bd729dab3a517b3ecc2d5/github/resource_github_repository.go#L684-L688)
- that fails with a 422 (unprocessable entity) because its trying to update the visibility from `internal` to `internal`
- the code [expects the error message to contain `422 Visibility is already internal`](https://github.com/integrations/terraform-provider-github/blob/e827c6f14ae0ca63327bd729dab3a517b3ecc2d5/github/resource_github_repository.go#L690-L691) and returns an error otherwise
- it appears that the GitHub api route has changed the 422 error message to `422 Visibility can't be changed by this user. and Visibility is already internal.` so the match fails.

**Steps to Reproduce**
Using the configuration above, run:

`terraform apply`

**Current Workaround**

As the resource is currently tainted (because [terraform taints resources when creation fails](https://developer.hashicorp.com/terraform/cli/state/taint#the-tainted-status)), rerunning terraform will delete/recreate the resource (because that's how terraform treats tainted resources) and run into the same issue. 
Since visibility is one of the last edited properties for the repository (meaning the repository creation has actually fully completed) it is currently appears to be safe to untaint the resource and reapply
```shell
terraform untaint github_repository.internal-repo-test
terraform apply
```

### PR DESCRIPTION
**Pull request by chapurlatn**
Resolves #1364

----

### Before the change?

* It ignores repo visibility update error and the visibility didn't actually changed.

### After the change?

* The same, but the error return by github changed and the ignore logic was broken.

### Other information

* 

----

### Pull request checklist
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Added the appropriate label for the given change

### Does this introduce a breaking change?

Please see our docs on [breaking changes](https://github.com/octokit/.github/blob/master/community/breaking_changes.md) to help!

- [ ] Yes (Please add the `Type: Breaking change` label)
- [x] No

If `Yes`, what's the impact:  

* N/A

### Pull request type

 
 

Please add the corresponding label for change this PR introduces:
- Bugfix: `Type: Bug`
- Feature/model/API additions: `Type: Feature`
- Updates to docs or samples: `Type: Documentation`
- Dependencies/code cleanup: `Type: Maintenance`

----

### CODE REVIEW

**Comment by chapurlatn** [comment]
See my comment in the issue for the error differences:
https://github.com/integrations/terraform-provider-github/issues/1364#issuecomment-1318427371

File: github/resource_github_repository.go [hunk 1]
```diff
@@ -687,7 +687,7 @@ func resourceGithubRepositoryUpdate(d *schema.ResourceData, meta interface{}) er
 		log.Printf("[DEBUG] Updating repository visibility from %s to %s", o, n)
 		_, _, err = client.Repositories.Edit(ctx, owner, repoName, repoReq)
 		if err != nil {
-			if !strings.Contains(err.Error(), fmt.Sprintf("422 Visibility is already %s", n.(string))) {
+			if !(strings.Contains(err.Error(), "422") && strings.Contains(err.Error(), fmt.Sprintf("Visibility is already %s", n.(string)))) {
```
[the remark this answers is not in the document]
Comment (chapurlatn) [reply github/resource_github_repository.go]: taken into account

**Review by jtgrohn — APPROVED** [review]

**Comment by Luis-3M** [comment]
@kfcampbell could you help approving this PR workflow ? 🙏

**Review by jhaumont — APPROVED** [review]
Thx for the PR !

**Comment by justin-kidman-axomic** [comment]
Any chance we can get this out?

**Review by nickfloyd — APPROVED** [review]
LGTM! ❤️

**Comment by nickfloyd** [comment]
Going out the door now: https://github.com/integrations/terraform-provider-github/releases/tag/v5.9.1

**Pull request merged**